New packages: tageditor, tagparser and c++utilities

ahesford opened the pull request:
This PR introduces packages for `tageditor` and its dependencies, `tagparser` and `c++utilities`. `tageditor` is a utility to manipulate tags on audio and video files. This build enables an optional qt5 GUI but provides build options to package a CLI-only version to keep dependencies light.

Chocimier commented on srcpkgs/tageditor/template:
```diff
@@ -0,0 +1,63 @@
+short_desc="Tag editor for AAC, ID3, Vorbis, Opus, FLAC and Matroska (CLI only)"
+maintainer="Andrew J. Hesford <ajh@sideband.org>"
+license="GPL-2.0-only"
+homepage="https://github.com/Martchus/${pkgname}"
```
Don't use variables in homepage, is should be copyable from template.

ahesford replied:
Done, here and in all other templates in the pull request

Chocimier commented on srcpkgs/tageditor/template:
```diff
@@ -0,0 +1,63 @@
+
+desc_option_webengine="Enable support for qt5-webengine file views (requires qt5)"
+desc_option_script="Enable support for qt5-script javascript (requires qt5)"
+desc_option_webkit="Enable support for qt5-webkit file views (requires qt5,~webengine)"
```
vopt_conflict

ahesford replied:
Thanks for the tip, vopt_conflict is now used.

Chocimier commented on srcpkgs/tageditor/template:
```diff
@@ -0,0 +1,63 @@
+desc_option_webkit="Enable support for qt5-webkit file views (requires qt5,~webengine)"
+
+case "$XBPS_TARGET_MACHINE" in
+	x86_64*|ppc64le*|aarch64*) build_options_default+=" webengine";;
```
Please copy condition from qt5 template.

ahesford replied:
Done

Chocimier commented on srcpkgs/tagparser/template:
```diff
@@ -0,0 +1,35 @@
+build_style=cmake
+configure_args="-DBUILD_SHARED_LIBS:BOOL=ON"
+makedepends="c++utilities-devel zlib-devel"
+depends="zlib"
```
it is detected:

=> tagparser-9.1.2_1: running pre-pkg hook: 04-generate-runtime-deps ...
   SONAME: libc++utilities.so.5 <-> c++utilities>=5.2.0_1
   SONAME: libz.so.1 <-> zlib>=1.2.3_1

ahesford replied:
Done

Chocimier commented on srcpkgs/tageditor/template:
```diff
@@ -0,0 +1,63 @@
+build_style=cmake
+configure_args="-DBUILD_SHARED_LIBS:BOOL=ON"
+makedepends="tagparser-devel c++utilities-devel"
+short_desc="Tag editor for AAC, ID3, Vorbis, Opus, FLAC and Matroska (CLI only)"
```
no more CLI only

ahesford replied:
Thanks for the catch, I missed this.

Chocimier commented:
Packages are usually named like tarball, not necessarily like content, so `cpp-utilities`. Otherwise looks ok. Build with webengine option passes.

ahesford commented:
> Packages are usually named like tarball, not necessarily like content, so `cpp-utilities`. Otherwise looks ok. Build with webengine option passes.

That was my original name for the package, but in reworking everything I moved to `c++utilities` because of the contents. I've reverted back to `cpp-utilities` as recommended.

Chocimier commented:
Please rebase

ahesford commented:
> Please rebase

Done, thanks!

Chocimier commented:
Tested, works. But please move changes to [proper commits](https://github.com/void-linux/void-packages/pull/19844/commits/1a88641dd9a4aaa4bfdbc4b998ebc6fa03a6badb#diff-816e5624e093eafd5fac2aa7b3b46e83R7), reorder so deps go before package using it, fix commit titles - no revision, remove `wrksrc="${pkgname}-${version}"` that is default.

ahesford commented:
> Tested, works. But please move changes to [proper commits](https://github.com/void-linux/void-packages/pull/19844/commits/1a88641dd9a4aaa4bfdbc4b998ebc6fa03a6badb#diff-816e5624e093eafd5fac2aa7b3b46e83R7), reorder so deps go before package using it, fix commit titles - no revision, remove `wrksrc="${pkgname}-${version}"` that is default.

Thanks for catching these mistakes. I've reworded the commit messages, moved the misplaced change to its proper commit (an artifact of an earlier rebase), and removed the unnecessary `wrksrc` definition (an artifact of the `c++utilities` to `cpp-utilities` rename).

The Github comment thread seems to be sorting the commit messages alphabetically, but the commits are ordered properly in `git log`. If you view history for commit [`926e663`](https://github.com/void-linux/void-packages/commits/926e663ace2a1ab81ae344cd52014115c757bb13/srcpkgs), you should see that the `qtutilities` commit is listed before the `tageditor` commit.

Chocimier commented:
Good job!
